### DISCUSSION TITLE
Deleting specific git commits

### DISCUSSION THREAD

**Discussion by donno2048** [community/community#22821] [New to GitHub]
<p>I have a repo with a file named main and in my committing history I committed several changes to it,<br>
now I want to delete it and every commit I have done to it, can I just delete every commit with the word ‘main’ in its name?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Most likely you’ll need to look at the file path rather than commit messages to do that, the <a href="https://git-scm.com/docs/git-filter-branch#_examples">examples for <code>git filter-branch</code></a> describe (among other things) how to remove a file from history. Alternatively, if the history isn’t too complex, you could use <a href="https://git-scm.com/docs/git-rebase"><code>git rebase</code></a> to manually edit the relevant commits.</p>
<p>Two words of caution though:</p>
<ol>
<li>Either method will change <strong>all</strong> commits starting from the first one that needs to be modified. So if your repository is shared with others, they’ll all notice the change and have to manually accept it (or not). In that case you should carefully consider if you really want to rewrite history.</li>
<li>If your repository is on Github and you’re trying to remove sensitive data, after you’ve fixed the repository you should contact support and ask them to make sure there are no caches or other references to the sensitive data left. See also: <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/removing-sensitive-data-from-a-repository">Removing sensitive data from a repository</a>
</li>
</ol>

**Reply by corneliusroemer** [reply]
How to do this?
>If your repository is on Github and you’re trying to remove sensitive data, after you’ve fixed the repository you should contact support and ask them to make sure there are no caches or other references to

**Reply by airtower-luna** [reply]
Hopefully here: https://support.github.com/request/remove-data

**Comment by donno2048** [comment]
<p>Thanks, this is very helpful</p>
